[waiter-palypoo] ip

waiter-palypoo opened the pull request.

waiter-palypoo committed:
Add HinaBot.java

waiter-palypoo committed:
Refactor greeting message and add handleCommand method to echo user input

waiter-palypoo committed:
Add code for adding items to a list and printing the list

waiter-palypoo committed:
Add Task class, add code to mark and unmark tasks

MingEn82 commented on src/main/java/HinaBot.java:
```diff
@@ -0,0 +1,7 @@
+        System.out.println("What are your orders?");
+        System.out.println("Goodbye master, let's meet again soon...");
+    }
+}
```
Nice!

MingEn82 commented on src/main/java/HinaBot.java:
```diff
@@ -1,7 +1,28 @@
+            System.exit(0);
+        }
     }
 }
```
Nice! No Coding Violations

MingEn82 commented on src/main/java/HinaBot.java:
```diff
@@ -17,12 +19,28 @@ private static void showGreeting() {
+            System.out.println(". " + taskList[i]);
+        }
     }
 }
```
Good job!

MingEn82 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,24 @@
+    public void setDescription(String description) {
+        this.description = description;
+    }
+}
```
Good job!

MingEn82 commented on src/main/java/HinaBot.java:
```diff
@@ -26,21 +26,49 @@ public static void handleCommand(String command) {
+        taskList[taskIndex - 1].setDone(false);
+        System.out.println("Roger that! This task is marked as done: ");
+        System.out.println("[ ] " + taskList[taskIndex - 1].getDescription());
+    }
```
No coding violations at all!

waiter-palypoo committed:
Add support for todo, deadline and event task types.

waiter-palypoo committed:
no message

waiter-palypoo committed:
Merge commit '7e758c3a91d41d43006c379c502de1ceb83a4e1f' into HEAD

# Conflicts:
#	src/main/java/HinaBot.java

waiter-palypoo committed:
Resolved error resulting from RCS mistake

waiter-palypoo committed:
Add code to handle unknown command, insufficient detail exceptions.

waiter-palypoo committed:
Merge branch 'branch-Level-5'

waiter-palypoo committed:
Divide classes into packages

bdthanh commented on src/main/java/hina/task/Task.java:
```diff
@@ -0,0 +1,37 @@
+        String mark;
+        if (isDone) {
+            mark = "X";
+        }
+        else {
```
else should be on the same line with }

bdthanh commented on src/main/java/hina/task/Event.java:
```diff
@@ -0,0 +1,24 @@
+        if (super.isDone()) {
+            mark = "X";
+        }
+        else {
```
Same problem with my comment in Task

bdthanh commented on src/main/java/hina/HinaBot.java:
```diff
@@ -0,0 +1,118 @@
+        System.out.println("What are your orders?");
+    }
+
+    public static void handleCommand(String command) throws HinaException {
+        if (command.equalsIgnoreCase("bye")) {
+            System.out.println("Goodbye master, let's meet again soon...");
+            System.exit(0);
+        } else if (command.equalsIgnoreCase("list")) {
+            listTasks();
+        } else if (command.split(" ")[0].equalsIgnoreCase("mark")) {
+            int taskIndex = Integer.parseInt(command.split(" ")[1]);
+            markTask(taskIndex);
+        } else if (command.split(" ")[0].equalsIgnoreCase("unmark")) {
+            int taskIndex = Integer.parseInt(command.split(" ")[1]);
+            unmarkTask(taskIndex);
+        } else if (command.split(" ")[0].equalsIgnoreCase("todo")) {
+            addTask(command.substring(5));
+        } else if (command.split(" ")[0].equalsIgnoreCase("deadline")) {
+            addDeadline(command.substring(9));
+        } else if (command.split(" ")[0].equalsIgnoreCase("event")) {
+            addEvent(command.substring(6));
+        } else {
+            throw new HinaException();
+        }
+    }
```
Good job of splitting a big method into smaller one and improving your code quality

bdthanh reviewed (commented):
Just a few coding violation in your code, good job! You can think of splitting the HinaBot into smaller classes that each one have a different responsibility. Good luck on your upcoming levels

waiter-palypoo committed:
Add method to delete tasks, fixed coding standard violations, change access levels in Task class

waiter-palypoo committed:
Add functionality to save task list to a .txt file.

waiter-palypoo committed:
Merge branch 'branch-Level-6'

waiter-palypoo committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/hina/HinaBot.java
#	src/main/java/hina/task/Event.java

waiter-palypoo committed:
Refactor functionality into Parser, Ui, Storage and TaskList classes.

Fixed bugs:
1. trailing whitespaces leading to storing blank information in Deadline and Event objects.
2. taskList now saves to the hard disk after marking and unmarking tasks.

waiter-palypoo committed:
Add date-time format support

waiter-palypoo committed:
Add method to search for items by matching substrings in Task descriptions in taskList.

waiter-palypoo committed:
Merge pull request #1 from waiter-palypoo/branch-Level-8

Add date-time format support

waiter-palypoo committed:
Merge branch 'master' of https://github.com/waiter-palypoo/ip

waiter-palypoo committed:
Merge branch 'master' into branch-Level-9

waiter-palypoo committed:
Merge pull request #2 from waiter-palypoo/branch-Level-9

Add method to search for items in taskList

waiter-palypoo committed:
Merge branch 'master' of https://github.com/waiter-palypoo/ip into branch-Level-9

waiter-palypoo committed:
Merge branch 'branch-Level-9'

waiter-palypoo committed:
Optimise imports for Deadline and Event classes

waiter-palypoo committed:
Add JavaDoc comments for multiple classes and methods. Refactor some command-line message outputs to methods in Ui class.

waiter-palypoo committed:
Merge pull request #3 from waiter-palypoo/branch-A-JavaDoc

Add JavaDoc documentation

waiter-palypoo committed:
Merge branch 'master' of https://github.com/waiter-palypoo/ip

waiter-palypoo committed:
Updated user guide, changed find method to print the index of matching Tasks in the TaskList

waiter-palypoo committed:
Fix bugs involving invalid inputs for mark, unmark and delete methods

waiter-palypoo committed:
Build jar file
